joplin/pages/topic_collection_page/factories.py

```diff
@@ -32,6 +33,64 @@ class Meta:
         model = JanisBasePageWithTopicCollections
 
     @factory.post_generation
-    def create_related_objects(self, create, extracted, **kwargs):
+    def add_topic_collections(self, create, extracted, **kwargs):
+        if extracted:
+            # A list of topic collections were passed in, use them
+            for topic_collection in extracted:
+                # todo: check to see if we already have the basepagetopiccollection objects made
+                JanisBasePageTopicCollectionFactory.create(page=self, topic_collection=topic_collection)
+            return
+
+        # todo figure out if this is really what we want this factory to do
         if create:
             JanisBasePageTopicCollectionFactory.create_batch(2, page=self)
+
+
+
+def create_topic_collection_page_from_page_dictionary(page_dictionary, revision_id):
```

Comment:
A suggestion to help readers follow the flow: calling this something like "create_topic_collection_page_from_importer" would let us know that these factory functions are being used by the importer process.